extensions/wikia/AdEngine/js/recovery/helper.js

```diff
@@ -0,0 +1,75 @@
+define('ext.wikia.adEngine.recovery.helper', [
+	'ext.wikia.adEngine.adContext',
+	'wikia.document',
+	'wikia.lazyqueue',
+	'wikia.log',
+	require.optional('ext.wikia.adEngine.provider.gpt.sourcePointTag')
+], function (
+	adContext,
+	doc,
+	lazyQueue,
+	log,
+	SourcePointTag
+) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.recovery.helper',
+		context = adContext.getContext(),
+		slotsToRecover = [],
+		onBlockingEventsQueue = [];
+
+	function initEventQueue() {
+		lazyQueue.makeQueue(onBlockingEventsQueue, function (eventHandler) {
+			doc.addEventListener('sp.blocking', function () {
+				eventHandler();
+			});
+		});
+		onBlockingEventsQueue.start();
+	}
+
+	function addSlotToRecover(slotName) {
+		slotsToRecover.push(slotName);
+	}
+
+	function addOnBlockingEvent(eventHandler) {
```

Comment:
Sure makes sense. Naming is always a tough task :japanese_ogre: I'll go with `addOnBlockingCallback` :)